core/dbt/graph/graph.py

```diff
@@ -59,18 +59,40 @@ def select_childrens_parents(self, selected: Set[UniqueId]) -> Set[UniqueId]:
     def select_children(
         self, selected: Set[UniqueId], max_depth: Optional[int] = None
     ) -> Set[UniqueId]:
-        descendants: Set[UniqueId] = set()
-        for node in selected:
-            descendants.update(self.descendants(node, max_depth))
-        return descendants
+        """Returns all nodes which are descendants of the 'selected' set.
+        Nodes in the 'selected' set are counted as children only if
+        they are descendants of other nodes in the 'selected' set."""
+        children: Set[UniqueId] = set()
+        i = 0
+        while len(selected) > 0 and (max_depth is None or i < max_depth):
+            next_layer: Set[UniqueId] = set()
+            for node in selected:
+                next_layer.update(self.graph.successors(node))
```

Comment:
It was my mistake. This line should read: `next_layer.update(self.descendants(node, 1))`

Reply:
I made this same mistake on first pass :P. Fixing!